rfd/0085-better-react-testing-remove-default-exports.md

```diff
@@ -0,0 +1,155 @@
+---
+authors: Ryan Clark (ryan.clark@goteleport.com)
+state: draft
+---
+
+# RFD 85 - Better React testing, remove default exports
+
+## What
+
+Improve the webapps codebase by changing the way we test our components and write our component stories.
+
+## Why
+
+The current convention is to use Storybook to render our large, full components in their different states.  This has a
+couple of knock-on effects.
+
+### Component Architecture
+
+As it's hard to mock data in Storybook, we end up exporting a component that is suitable for rendering in Storybook,
+such as:
+
+```typescript jsx
+export function DesktopSession(props: State) {
+  const {
+    directorySharingState,
+    setDirectorySharingState,
+    clipboardState,
+    fetchAttempt,
+    tdpConnection,
+    disconnected,
+    wsConnection,
+    setTdpConnection,
+  } = props;
+  // etc
+
+  return (
+    <Flex flexDirection="column">
+      // etc
+    </Flex>
+  );
+}
+```
+
+This component takes all of its state and business logic from the props passed in.
+
+For use in the actual application, we wrap the component above in a default export which populates the props the
+component needs by using a hook.
+
+```typescript jsx
+export default function Container() {
+  const state = useDesktopSession();
+  return <DesktopSession {...state} />;
+}
+```
+
+
+### Testing
+
+As a result of the above, we end up testing the `DesktopSession` component instead of the actual component (`Container`) that is
```

Comment:
Try to not reference by location in the document, this causes issues with restructuring the content. 

The `Component Architecture` section is saying that we write it that way because of testing, but now the `Testing` section is saying we write the tests because of the `Component Architecture`.

Maybe this is a better place to mention the storybook content from above.

It's probably worth adding something to the effect that we're using Storybook for something it wasn't intended for causing these issues.